### PR TITLE
RUN-446: auth resource access refactor

### PR DESCRIPTION
**Pull request by gschueler**
**Is this a bugfix, or an enhancement? Please describe.**
WIP: Adds new mechansim for authorized access to application resources.

**Describe the solution you've implemented**

**Describe alternatives you've considered**

**Additional context**
Add any other context or screenshots about the change here.

### CODE REVIEW

File: rundeck-authz/rundeck-authz-core/src/main/java/org/rundeck/core/auth/access/AuthActionsUtil.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Utility methods for AuthActions interface
+ */
+public class AuthActionsUtil {
+
```
Comment (imad6639) [inline rundeck-authz/rundeck-authz-core/src/main/java/org/rundeck/core/auth/access/AuthActionsUtil.java:16]: Probably worth it to add a non public constructor to these Util classes so java doesnt add an implicit public constructor

File: rundeckapp/grails-app/services/rundeck/services/ApiService.groovy [hunk 2]
```diff
@@ -366,26 +369,15 @@ class ApiService {
...
-    }
-
-    @CompileStatic
-    def appendResponseOutput(HttpServletResponse response, String output) {
-        response.outputStream << output
-    }
+//    @CompileStatic
+//    def appendResponseOutput(HttpServletResponse response, String output) {
+//        response.outputStream << output
+//    }
```
Comment (imad6639) [inline rundeckapp/grails-app/services/rundeck/services/ApiService.groovy:376]: lets delete this

**Review by imad6639 — APPROVED** [review]

File: rundeckapp/grails-app/services/rundeck/services/ApiService.groovy (hunk 2 above)
Comment (gschueler) [inline rundeckapp/grails-app/services/rundeck/services/ApiService.groovy:376]:
```suggestion
```

File: rundeckapp/grails-app/services/rundeck/services/ApiService.groovy [hunk 3]
```diff
@@ -366,26 +369,11 @@ class ApiService {
...
-    def appendResponseOutput(HttpServletResponse response, String output) {
-        response.outputStream << output
-    }
 
-    def respondXml(HttpServletResponse response, Closure recall) {
-        return respondOutput(response, APPLICATION_XML_CONTENT_TYPE, renderXml(recall))
-    }
+//    def respondXml(HttpServletResponse response, Closure recall) {
+//        return respondOutput(response, APPLICATION_XML_CONTENT_TYPE, renderXml(recall))
+//    }
```
Comment (gschueler) [inline rundeckapp/grails-app/services/rundeck/services/ApiService.groovy:376]:
```suggestion
```

**Pull request merged**